rmgpy/molecule/molecule.py

```diff
@@ -1595,4 +1595,66 @@ def toGroup(self):
         
         return group
 
-
+    def getFirstNeighbor(self, atom):
```

Comment:
All these getNeighbor functions seem like they should be done recursively. Just as quick pseudo-code suggestion.

def recursivelyGetNthNeighbors( startingAtoms, n, blacklist):
   if startingAtoms is not a list: [startingAtoms] 
   neighbors = []
   for atom in startAtoms:
       neighbors.extend([neighbor if neighbor.isNonHydrogen() for neighbor in self.getBonds(atom)])
   neighbors = list(set(neighbors)-set(blackList)) #remove duplicates and atoms already covered in blacklist
   if n >0:
     neighbors = recursivelyGetNthNeighbors(neighbors, n-1, startingAtoms)
   return neighbors

Above pseudo-code might need polish/debugging, but you get the idea.